docs/features/key_cancellation.md

```diff
@@ -0,0 +1,65 @@
+# Key Cancellation
+
+This feature iterates over the `key_cancellation_list`, and upon any matches of key down, all specified keys are released.
+
+Imagine you're holding down the `A` key. If you then press the `D` key, the `A` key will stop being active. To use the `A` key again, you need to press it once more.
+
+## Usage {#usage}
+
+Add the following to your `rules.mk`
+```make
+KEY_CANCELLATION_ENABLE = yes
+```
+
+By default, key cancellation is disabled even after adding to `rules.mk`. To enable it, you need to use the `KX_CATG` or `KX_CAON` keycode to enable it. The status is stored in persistent memory, so you shouldn't need to enable it again.
+
+Your `keymap.c` will then need a Key Cancellation list definition:
+
+```c
+const key_cancellation_t PROGMEM key_cancellation_list[] = {
+    // on key down
+    //       |    key to be released
+    //       |     |
+    [0] = {KC_D, KC_A},
+    [1] = {KC_A, KC_D},
+};
+```
+
```

Comment:
Per previous suggestions, this should be moved so its not between the API sections, here or at the end of the file logically works best.
````suggestion

## Examples {#examples}
### SOCD WASD

```c
const key_cancellation_t PROGMEM key_cancellation_list[] = {
    [0] = {KC_D, KC_A},
    [1] = {KC_A, KC_D},
    [2] = {KC_W, KC_S},
    [3] = {KC_S, KC_W},
};
```
````